src/common/libidset/idset.c

```diff
@@ -394,8 +394,6 @@ bool idset_equal (const struct idset *idset1,
 
     if (!idset1 || !idset2)
         return false;
-    if (idset_count (idset1) != idset_count (idset2))
-        return false;
```

Comment:
This looks like it was an optimization, even though unnecessary. It might be helpful to mention in the commit message why the optimization wasn't needed or was a false optimization.

Reply:
i.e. probably because it is slow when IDSET_FLAG_COUNT_BY_ITER is set?

Reply:
OK, I'll work on that description.  Yes it was going to become inefficient.  In fact, now that you mention it, maybe it should remain, but be skipped if that flag is set.  With a comment inline.